src/log/log.c

```diff
@@ -2090,6 +2092,18 @@ __wt_log_release(WT_SESSION_IMPL *session, WT_LOGSLOT *slot, bool *freep)
 	if (F_ISSET(slot, WT_SLOT_CLOSEFH))
 		__wt_cond_signal(session, conn->log_file_cond);
 
+	if (F_ISSET(slot, WT_SLOT_SYNC_DIRTY) && !F_ISSET(slot, WT_SLOT_SYNC) &&
+	    conn->log_dirty_max != 0) {
```

Comment:
I think checking the `WT_SLOT_SYNC_DIRTY` flag is sufficient and we don't also need to check `conn->log_dirty_max`. That cannot be set with a 0 `log_dirty_max` (and if it changes via reconfigure, that's okay if this one falls on the side of syncing).